packages/btcindexer/src/btcindexer.ts

```diff
@@ -446,4 +454,31 @@ export class Indexer implements Storage {
 			};
 		});
 	}
+
+	async registerBroadcastedNbtcTx(
+		txHex: string,
+	): Promise<{ tx_id: string; registered_deposits: number }> {
+		const tx = Transaction.fromHex(txHex);
+		const txId = tx.getId();
+
+		const deposits = this.findNbtcDeposits(tx);
+		if (deposits.length === 0) {
+			throw new Error("Transaction does not contain any valid nBTC deposits.");
+		}
+
+		const now = Date.now();
+		const insertStmt = this.d1.prepare(
+			`INSERT OR IGNORE INTO nbtc_minting (tx_id, vout, sui_recipient, amount_sats, status, created_at, updated_at)
+         VALUES (?, ?, ?, ?, 'broadcasting', ?, ?)`,
+		);
+
+		const statements = deposits.map((deposit) =>
+			insertStmt.bind(txId, deposit.vout, deposit.suiRecipient, deposit.amountSats, now, now),
+		);
+
+		await this.d1.batch(statements);
+
+		console.log(`Successfully registered ${statements.length} deposit(s) for nBTC tx ${txId}.`);
+		return { tx_id: txId, registered_deposits: statements.length };
```

Comment:
Any reason we use tx_id and registered_deposits more than txId and registeredDeposits